Add mac os x install instructions

For optional, not-pure-python option, as they are not covered in the pypoman install instructions.

## work log

Run

```console

```

Get error

```console
✗ poetry add 'fake-data-for-learning[probability_polytope]'
Using version ^0.4.4 for fake-data-for-learning

Updating dependencies
Resolving dependencies... (0.6s)

Package operations: 4 installs, 0 updates, 0 removals

  • Installing pycddlib (3.0.2): Failed

  ChefBuildError

  Backend subprocess exited when trying to invoke build_wheel

  running bdist_wheel
  running build
  running build_py
  running egg_info
  writing src/pycddlib.egg-info/PKG-INFO
  writing dependency_links to src/pycddlib.egg-info/dependency_links.txt
  writing requirements to src/pycddlib.egg-info/requires.txt
  writing top-level names to src/pycddlib.egg-info/top_level.txt
  reading manifest file 'src/pycddlib.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  no previously-included directories found matching 'test'
  adding license file 'LICENSE.md'
  adding license file 'AUTHORS'
  writing manifest file 'src/pycddlib.egg-info/SOURCES.txt'
  creating build/lib.macosx-15.0-arm64-cpython-312/cdd
  copying src/cdd/__init__.pyi -> build/lib.macosx-15.0-arm64-cpython-312/cdd
  copying src/cdd/gmp.pyi -> build/lib.macosx-15.0-arm64-cpython-312/cdd
  copying src/cdd/py.typed -> build/lib.macosx-15.0-arm64-cpython-312/cdd
  running build_ext
  Compiling cython/_cdd.pyx because it changed.
  [1/1] Cythonizing cython/_cdd.pyx
  building 'cdd.__init__' extension
  creating build/temp.macosx-15.0-arm64-cpython-312/cython
  clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk -I/opt/homebrew/opt/openblas/include -I/private/var/folders/4j/sbkhdmj513bdc84vf485j54c0000gn/T/tmp37iuv42o/.venv/include -I/opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/include/python3.12 -c cython/_cdd.c -o build/temp.macosx-15.0-arm64-cpython-312/cython/_cdd.o
  cython/_cdd.c:1248:10: fatal error: 'cddlib/setoper.h' file not found
   1248 | #include "cddlib/setoper.h"
        |          ^~~~~~~~~~~~~~~~~~
  1 error generated.
  error: command '/usr/bin/clang' failed with exit code 1


  at ~/Library/Application Support/pypoetry/venv/lib/python3.11/site-packages/poetry/installation/chef.py:164 in _prepare
      160│
      161│                 error = ChefBuildError("\n\n".join(message_parts))
      162│
      163│             if error is not None:
    → 164│                 raise error from None
      165│
      166│             return path
      167│
      168│     def _prepare_sdist(self, archive: Path, destination: Path | None = None) -> Path:

Note: This error originates from the build backend, and is likely not a problem with poetry but with pycddlib (3.0.2) not supporting PEP 517 builds. You can verify this by running 'pip wheel --no-cache-dir --use-pep517 "pycddlib (==3.0.2)"'.
```

Following now: https://github.com/mcmtroffaes/pycddlib/issues/81

Then in poetry shell, I can pip install pycddlib via

```console
env "CFLAGS=-I$(brew --prefix)/include -L$(brew --prefix)/lib" python -m pip install pycddlib
```

Can restore with poetry by adding to `.zshrc`

```bash
export CFLAGS="-I$(brew --prefix)/include -L$(brew --prefix)/lib"
```

So now related .zshrc .bashrc looks like

```bash
# openblas
export LDFLAGS="-L/opt/homebrew/opt/openblas/lib"
export CPPFLAGS="-I/opt/homebrew/opt/openblas/include"

# pycddlib
export CFLAGS="-I$(brew --prefix)/include -L$(brew --prefix)/lib"
export PATH="/opt/homebrew/opt/cython/bin:$PATH"

# For pkg-config to find openblas you may need to set:
export PKG_CONFIG_PATH="/opt/homebrew/opt/openblas/lib/pkgconfig"
```

--- a/README.md
+++ b/README.md
@@ -124,6 +124,10 @@ This package exists because I became tired of googling for existing implementati
 
 ## Change log
 
+### v0.4.5
+
+Add to Mac OS X optional install instructions.
+
 ### v0.4.4
 
 Fix missing usage of optional dependency specification

--- a/README.md
+++ b/README.md
@@ -72,9 +72,32 @@ Install from [pypi](https://pypi.org/project/fake-data-for-learning/): `pip inst
 
 ### Optional dependencies
 
-Note that the methods of `utils.ProbabilityPolytope` that use polytope calculatations to generate conditional probability tables subject to constraints on expectation value use the non-pure-python library [pypoman](https://github.com/stephane-caron/pypoman). See the [installation instructions](https://github.com/stephane-caron/pypoman#installation) for external dependency instructions.
+Note that the methods of `utils.ProbabilityPolytope` that use polytope calculations to generate conditional probability tables subject to constraints on expectation value require the non-pure-python library [pypoman](https://github.com/stephane-caron/pypoman). System dependencies include [`cython`](https://cython.org) and [`glpk`](https://www.gnu.org/software/glpk/). See the [installation instructions](https://github.com/stephane-caron/pypoman#installation) for external dependency instructions.
 
-By default the dependencies for `utils.ProbabilityPolytope` are not installed; to do so, run from your virtual environment `pip install 'fake-data-for-learning[probability_polytope]'`
+On Mac OS X, these can be installed with
+
+```console
+brew install cython glpk cddlib gmp
+```
+
+(see also the [pycddlib](https://github.com/mcmtroffaes/pycddlib) issue [Installation using pip on MacOS 11.6 #49](https://github.com/mcmtroffaes/pycddlib/issues/49), but beware the typo in the first comment dependency name).
+
+Follow the instructions regarding setting environment variables (e.g. in your `.bashrc` or `.zshrc` files). For example, your `.zshrc` file on Mac OS X might contain
+
+```bash
+# openblas
+export LDFLAGS="-L/opt/homebrew/opt/openblas/lib"
+export CPPFLAGS="-I/opt/homebrew/opt/openblas/include"
+
+# pycddlib
+export CFLAGS="-I$(brew --prefix)/include -L$(brew --prefix)/lib"
+export PATH="/opt/homebrew/opt/cython/bin:$PATH"
+
+# For pkg-config to find openblas you may need to set:
+export PKG_CONFIG_PATH="/opt/homebrew/opt/openblas/lib/pkgconfig"
+```
+
+By default the python dependencies for `utils.ProbabilityPolytope` are not installed; to do so, run from your virtual environment `pip install 'fake-data-for-learning[probability_polytope]'`
 
 ### Local development
 

--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ extras_require = {
 
 setup(
     name='fake_data_for_learning',
-    version='0.4.4',
+    version='0.4.5',
     long_description=description,
     long_description_content_type='text/markdown',
     project_urls={

